### ISSUE TITLE
Rule proposal: `require-reduce-initial-value`

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2454]
### Description

 If an array is empty, calling `reduce`/`reduceRight` method will cause runtime error. We need a rule to force passing initial value to `reduce`/`reduceRight` method
```ts
const nums: number[] = [];
const sum = nums.reduce((result, num) => result + num); // This will cause runtime error
```

### Fail

```js
[].reduce((a, b)=>a+b);
[].reduceRight((a, b)=>a+b);
someArray.reduce((a, b)=>a+b);
someArray.reduceRight((a, b)=>a+b);
```

### Pass

```js
[].reduce((a, b)=>a+b,0);
[].reduceRight((a, b)=>a+b,foo);
someArray.reduce((a, b)=>a+b, 0);
someArray.reduceRight((a, b)=>a+b, foo);
Array.prototype.reduce.call(nums, (x,y)=>x+y);
Array.prototype.reduce.appy(nums, [(x,y)=>x+y]);
```

### Proposed rule name

require-reduce-initial-value
